lib/Filters/GitDiffReader.cpp

```diff
@@ -0,0 +1,76 @@
+#include "mull/Filters/GitDiffReader.h"
+
+#include "mull/Diagnostics/Diagnostics.h"
+#include <mull/Path.h>
+#include "mull/Toolchain/Runner.h"
+
+#include <iostream>
+#include <regex>
+#include <sstream>
+
+using namespace mull;
+
+GitDiffReader::GitDiffReader(Diagnostics &diagnostics, const std::string gitRepoPath)
+    : diagnostics(diagnostics), gitRepoPath(gitRepoPath) {}
+
+GitDiffInfo GitDiffReader::readGitDiff(const std::string &gitBranch) {
+  /// The implementation is borrowed from the git-clang-format Python tool.
+  /// https://opensource.apple.com/source/clang/clang-800.0.38/src/tools/clang/tools/clang-format/git-clang-format.auto.html
+  Runner runner(diagnostics);
+
+  std::vector<std::string> args = { "diff", "-U0", gitBranch };
+  std::vector<std::string> env = {};
+
+  ExecutionResult result = runner.runProgram("git", args, env, 1000, true);
```

Comment:
1 second seems to be too optimistic. Shall we make it configurable as well?

Reply:
Also, don't you want to `cd` into `gitRepoPath` first? Since we require this piece it makes sense to not require running Mull from the git repo directory.

Reply:
How about simply increasing this one to `5000`? It is a one-time action so should be safe enough.

Reply:
> Also, don't you want to cd into gitRepoPath first? 

Good catch. I will do it.